app/stac_api/validators.py

```diff
@@ -364,8 +364,20 @@ def validate_geometry(geometry):
     return geometry
 
 
-def validate_item_properties_datetimes_dependencies(
-    properties_datetime, properties_start_datetime, properties_end_datetime
+def validate_datetime_format(date_string):
+    try:
+        if not isinstance(date_string, datetime) and date_string is not None:
+            return fromisoformat(date_string)
+        return date_string
+    except ValueError as error:
+        logger.error("Invalid datetime string %s", error)
+        raise ValidationError(
+            _('Invalid datetime string %(error)s'), params={'error': error}, code='invalid'
+        ) from error
+
+
+def validate_item_properties_datetimes(
+    properties_datetime, properties_start_datetime, properties_end_datetime, properties_expires
```

Comment:
Just some higher-level thinking here, (it doesn't have to do with your change directly):
I think it might be better if the callers of this function made sure that what they pass in is either `datetime` or `None`. Instead, the above function, as a side-effect, parses the dates in case they are strings, even though this here is about validation... (I guess that's a result of not having strict typing 😁)

Reply:
I played around with different options, but nothing that is really nicer.